x-pack/plugins/cases/public/components/case_view/case_view_page.tsx

```diff
@@ -0,0 +1,447 @@
+        fetchCaseMetrics();
+      },
+      [updateCase, fetchCaseUserActions, caseId, subCaseId, fetchCaseMetrics]
+    );
+
+    const handleUpdateField = useCallback(
+      (newCase: Case, updateKey: UpdateKey) => {
+        updateCase({ ...newCase, comments: caseData.comments });
+        fetchCaseUserActions(caseId, newCase.connector.id, subCaseId);
+        if (updateKey === 'connector' || updateKey === 'status') {
+          fetchCaseMetrics();
+        }
+      },
+      [updateCase, caseData, fetchCaseUserActions, caseId, subCaseId, fetchCaseMetrics]
+    );
+
+    const { onUpdateField, isLoading, loadingKey } = useOnUpdateField({
+      caseId,
+      subCaseId,
+      caseData,
+      handleUpdateField,
+    });
+
+    // Set `refreshRef` if needed
+    useEffect(() => {
+      let isStale = false;
+      if (refreshRef) {
+        refreshRef.current = {
+          refreshCase: async () => {
+            // Do nothing if component (or instance of this render cycle) is stale or it is already loading
+            if (isStale || isLoading || isLoadingMetrics || isLoadingUserActions) {
+              return;
+            }
+            await Promise.all([fetchCase(true), fetchCaseMetrics(true), refetchCaseUserActions()]);
+          },
+        };
+        return () => {
+          isStale = true;
+          refreshRef.current = null;
+        };
+      }
+    }, [
+      fetchCase,
+      fetchCaseMetrics,
+      refetchCaseUserActions,
+      isLoadingUserActions,
+      isLoadingMetrics,
+      isLoading,
+      refreshRef,
+      updateCase,
+    ]);
+
+    const {
+      loading: isLoadingConnectors,
+      connectors,
+      permissionsError,
+    } = useConnectors({
+      toastPermissionsErrors: false,
+    });
+
+    const [connectorName, isValidConnector] = useMemo(() => {
+      const connector = connectors.find((c) => c.id === caseData.connector.id);
+      return [connector?.name ?? '', !!connector];
+    }, [connectors, caseData.connector]);
+
+    const currentExternalIncident = useMemo(
+      () =>
+        caseServices != null && caseServices[caseData.connector.id] != null
+          ? caseServices[caseData.connector.id]
+          : null,
+      [caseServices, caseData.connector]
+    );
+
+    const onSubmitConnector = useCallback(
+      (connectorId, connectorFields, onError, onSuccess) => {
+        const connector = getConnectorById(connectorId, connectors);
+        const connectorToUpdate = connector
+          ? normalizeActionConnector(connector)
+          : getNoneConnector();
+
+        onUpdateField({
+          key: 'connector',
+          value: { ...connectorToUpdate, fields: connectorFields },
+          onSuccess,
+          onError,
+        });
+      },
+      [onUpdateField, connectors]
+    );
+
+    const onSubmitTags = useCallback(
+      (newTags) => onUpdateField({ key: 'tags', value: newTags }),
+      [onUpdateField]
+    );
+
+    const onSubmitTitle = useCallback(
```

Comment:
Notes for another PR: I think we should move each of these `onSubmitTags, onSubmitTitle, ...` inside the corresponding components. The components can use the `useOnUpdateField` which should be moved outside of the `CaseViewPage` component.

Reply:
Yes it makes sense 👍  I will leave that for a following PR